src/components/dashboard/Dashboard.js

```diff
@@ -169,59 +172,49 @@ class Dashboard extends Component {
         this.setState({ users: tempArr })
     }
 
-    handleEditCategory = (event, users, categoryName) => {
+    handleEditCategory = async (event, users, categoryName) => {
         event.preventDefault();
 
         let editCategoryId = JSON.parse(localStorage.getItem('editCategoryId'));
-        let catStatesUsernames = [];
-        let catStatesName = '';
-        for(let i = 0; i < this.state.categories.length; i++) {
-            if(editCategoryId === this.state.categories[i].id) {
-                catStatesUsernames = this.state.categories[i].username;
-                catStatesName = this.state.categories[i].category_name;
+        let userIdArr = [this.user.userId];
+
+        if(users.length > 0) {
+            let res = await fetch(`${baseUrl}/get-users`);
+            let userTable = await res.json();
+            console.log(userTable);
+            for(let x = 0; x < users.length; x++) {
+                for(let y = 0; y < userTable.length; y++) {
+                    if(users[x].toLowerCase() === userTable[y].username.toLowerCase()) {
+                        userIdArr.push(userTable[y].id);
+                    } 
+                }
             }
+            // removes any duplicates in array
+            userIdArr = [...new Set(userIdArr)];
         }
 
-        if(categoryName !== catStatesName || users !== catStatesUsernames) {
-            let userIdArr = [this.user.userId];
-            if(users.length > 0) {
-                fetch(`${baseUrl}/get-users`)
-                .then(res => res.json())
-                .then(userTable => {
-                    for(let x = 0; x < users.length; x++) {
-                        for(let y = 0; y < userTable.length; y++) {
-                            if(users[x].toLowerCase() === userTable[y].username.toLowerCase()) {
-                                userIdArr.push(userTable[y].id);
-                            } 
-                        }
-                    }
-                    // removes any duplicates in array
-                    userIdArr = [...new Set(userIdArr)];
-                    fetch(`${baseUrl}/update-category/${editCategoryId}`, {
-                        method: 'PUT',
-                        headers: {
-                            'Content-Type': 'application/json'
-                        },
-                        body: JSON.stringify({
-                            categoryName: categoryName,
-                            userIds: userIdArr
-                        })
-                    })
-                    .then(res => {
-                        this.getCategories();
-                    })
-                    .catch(err => {
-                        console.log(err);
-                    })
-                    this.setState({ 
-                        categoryName: categoryName,
-                        users: userIdArr
-                    });
+        if(this.OGCategoryName !== categoryName) {
```

Comment:
Does it work if you only change a user and not the category name?

Reply:
I changed it, if you want to look at it again.